test/unit/org/apache/cassandra/index/sai/plan/FlushIndexWhileQueryingTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.plan;
+
+import java.util.concurrent.ForkJoinPool;
+
+import org.apache.cassandra.index.sai.SAITester;
+import org.apache.cassandra.inject.Injections;
+import org.apache.cassandra.inject.InvokePointBuilder;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test to cover edge cases related to memtable flush during query execution.
+ */
+public class FlushIndexWhileQueryingTest extends SAITester
+{
+    @Test
+    public void testFlushAfterQueryViewBuildEqualityQuery() throws Throwable
+    {
+        createTable("CREATE TABLE %s (k text PRIMARY KEY, x int)");
+
+        createIndex("CREATE CUSTOM INDEX ON %s(x) USING 'StorageAttachedIndex'");
+        waitForTableIndexesQueryable();
+
+        execute("INSERT INTO %s (k, x) VALUES (?, ?)", "a", 0);
+        execute("INSERT INTO %s (k, x) VALUES (?, ?)", "b", 0);
+        execute("INSERT INTO %s (k, x) VALUES (?, ?)", "c", 1);
+
+        testFlushAfterQueryViewBuiltBeforeQueryExecution("SELECT k FROM %s WHERE x = 0", 2);
+    }
+
+    @Test
+    public void testFlushAfterQueryViewBuildNotContainsQuery() throws Throwable
+    {
+        createTable("CREATE TABLE %s (k text PRIMARY KEY, x set<int>)");
+
+        createIndex("CREATE CUSTOM INDEX ON %s(x) USING 'StorageAttachedIndex'");
+        waitForTableIndexesQueryable();
```

Comment:
As Caleb pointed out in the Apache PR, we do not need this. https://github.com/datastax/cassandra/blob/main/test/unit/org/apache/cassandra/cql3/CQLTester.java#L1017
I forgot about that before, sorry.

Reply:
Done